examples/contrib/tracking/tracking_svi_em.py

```diff
@@ -0,0 +1,286 @@
+        exists = pyro.sample("exists", assignment.exists_dist, infer={"enumerate": "parallel"})
+        with poutine.scale(scale=exists):
+            #  states_var = states_cov.reshape(states_cov.shape[:-2] + (-1,))[..., ::n+1]
+            #  pyro.sample("states", dist.Normal(states_loc, states_var).independent(1))
+            pyro.sample("states", dist.Delta(states_loc, event_dim=1))
+    with pyro.iarange("detections", observations.shape[1]):
+        with poutine.scale(scale=is_observed.float()):
+            with pyro.iarange("time", args.num_frames):
+                pyro.sample("assign", assignment.assign_dist, infer={"enumerate": "parallel"})
+
+    return assignment, states_loc
+
+
+def main(args):
+    if isinstance(args, str):
+        args = make_args(args)
+
+    viz = init_visdom(args.visdom)
+    pyro.set_rng_seed(0)
+    true_states, true_positions, observations = generate_observations(args)
+    true_num_objects = len(true_states)
+    max_num_detections = observations.shape[1]
+    assert true_states.shape == (true_num_objects, 2)
+    assert true_positions.shape == (args.num_frames, true_num_objects)
+    assert observations.shape == (args.num_frames, max_num_detections, 2)
+    print("generated {:d} detections from {:d} objects".format(
+        (observations[..., -1] > 0).long().sum(), true_num_objects))
+    print('true_states = {}'.format(true_states))
+
+    pyro.set_rng_seed(1)  # Use a different seed from data generation
+    pyro.clear_param_store()
+    assignment, states_loc = guide(args, observations)
+    p_exists = assignment.exists_dist.probs
+    positions = get_positions(states_loc, args.num_frames)
+    plot_solution(observations, p_exists, positions, true_positions, args, 'after 10 EM (with merge)', viz=viz)
+    plot_exists_prob(p_exists, viz)
+
+    pyro.set_rng_seed(1)  # Use a different seed from data generation
+    pyro.clear_param_store()
+
+    elbo = TraceEnum_ELBO(max_iarange_nesting=2)
+    optim = ClippedAdam({'lr': 0.1})
+    svi = SVI(model, guide, optim, elbo)
+    losses = []
+    for epoch in range(args.svi_iters):
+        loss = svi.step(args, observations)
+        losses.append(loss)
+        print('epoch {: >3d} loss = {}, emission_noise_scale = {}'.format(
+            epoch, loss, pyro.param("emission_noise_scale").item()))
+    if viz is not None:
+        viz.plot(losses)
+    else:
+        pyplot.figure().patch.set_color('white')
+        pyplot.plot(losses)
+
+    assignment, states_loc = guide(args, observations)
+    p_exists = assignment.exists_dist.probs
+    positions = get_positions(states_loc, args.num_frames)
+    plot_solution(observations, p_exists, positions, true_positions, args,
+                  'after 10 EM (with prune and merge)', viz=viz)
+    plot_exists_prob(p_exists, viz)
+
+
+def arg_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--num_frames', default=40, type=int, help='number of frames')
```

Comment:
nit: argparsae convention is to use dashes rather than underscores, and let the library automatically convert dashes to underscores. So you can write
```py
parser.add_argument('--max-num-objects', ...)
```
and still use python identifiers on the result
```py
args.max_num_objects
```